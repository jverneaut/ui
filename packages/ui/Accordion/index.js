import AccordionCore from './Accordion.js';
import AccordionItem from './AccordionItem.js';

/**
 * @typedef {import('./Accordion.js').AccordionInterface} AccordionInterface
 * @typedef {import('./AccordionItem.js').AccordionItemInterface} AccordionItemInterface
 */

/**
 * Accordion class.
 */
class Accordion extends AccordionCore {
  static config = {
    ...AccordionCore.config,
    components: {
      AccordionItem,
    },
  };
}

export default Accordion;
