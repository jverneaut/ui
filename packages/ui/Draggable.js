import styler from 'stylefire';
import { Base, withDrag } from '@studiometa/js-toolkit';

/**
 * @typedef {import('stylefire').Styler} Styler
 * @typedef {import('@studiometa/js-toolkit/services/drag.js').DragServiceProps} DragServiceProps
 */

/**
 * @typedef {Draggable} DraggableInterface
 */

/**
 * Draggable class.
 *
 * @link https://jsfiddle.net/soulwire/znj683b9/
 */
export default class Draggable extends withDrag(Base) {
  static config = {
    name: 'DraggableElement',
  };

  /**
   * The origin of the element.
   * @type {{ x: number, y: number }}
   */
  elementOrigin = {
    x: 0,
    y: 0,
  };

  /**
   * The element styler for performant style update.
   * @type {Styler}
   */
  styler;

  /**
   * Mounted hook.
   * @return {void}
   */
  mounted() {
    this.styler = styler(this.$el);
  }

  /**
   * Destroyed hook.
   * @return {void}
   */
  destroyed() {
    this.styler = undefined;
  }

  /**
   * Drag service hook.
   * @param {DragServiceProps} props
   */
  dragged(props) {
    if (props.mode === 'start') {
      this.elementOrigin.x = this.styler.get('x');
      this.elementOrigin.y = this.styler.get('y');
      return;
    }

    this.styler.set({
      x: this.elementOrigin.x + props.x - props.origin.x,
      y: this.elementOrigin.y + props.y - props.origin.y,
    });
  }
}
