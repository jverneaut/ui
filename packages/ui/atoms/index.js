export { default as Cursor } from './Cursor/Cursor.js';
export { default as Figure } from './Figure/Figure.js';
export { default as LazyInclude } from './LazyInclude/LazyInclude.js';
