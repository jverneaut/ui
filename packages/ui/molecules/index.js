export * from './TableOfContent/index.js';
export { default as Accordion } from './Accordion/Accordion.js';
export { default as AccordionItem } from './Accordion/AccordionItem.js';
export { default as Modal } from './Modal/Modal.js';
export { default as ModalWithTransition } from './Modal/ModalWithTransition.js';
export { default as Panel } from './Panel/Panel.js';
export { default as Slider } from './Slider/Slider.js';
export { default as SliderItem } from './Slider/SliderItem.js';
export { default as Sticky } from './Sticky/Sticky.js';
export { default as Tabs } from './Tabs/Tabs.js';
