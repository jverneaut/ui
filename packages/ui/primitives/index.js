// eslint-disable-next-line import/prefer-default-export
export { default as Draggable } from './Draggable/Draggable.js';
export { default as Sentinel } from './Sentinel/Sentinel.js';
