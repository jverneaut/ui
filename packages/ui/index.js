import Accordion from './Accordion/Accordion.js';
import Cursor from './Cursor/Cursor.js';
import Draggable from './Draggable/Draggable.js';
import Figure from './Figure/Figure.js';
import Modal from './Modal/Modal.js';
import Tabs from './Tabs/Tabs.js';

export { Accordion, Cursor, Draggable, Figure, Modal, Tabs };
