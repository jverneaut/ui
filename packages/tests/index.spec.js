import * as components from '@studiometa/ui';
import getFilenamesInFolder from './__utils__/getFilenamesInFolder.js';

test('components exports', () => {
  const names = getFilenamesInFolder('../ui/', import.meta.url).filter(
    (name) => !name.endsWith('.json') && name !== '.DS_Store' && name !== 'node_modules'
  );
  expect(Object.keys(components)).toEqual(names);
});
