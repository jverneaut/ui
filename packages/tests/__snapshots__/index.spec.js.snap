// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components exports 1`] = `
Array [
  "AbstractPrefetch",
  "AbstractScrollAnimation",
  "AbstractSliderChild",
  "Accordion",
  "AccordionItem",
  "AnchorScrollTo",
  "Cursor",
  "Draggable",
  "Figure",
  "LargeText",
  "LazyInclude",
  "Menu",
  "MenuBtn",
  "MenuList",
  "Modal",
  "ModalWithTransition",
  "Panel",
  "PrefetchWhenOver",
  "PrefetchWhenVisible",
  "ScrollAnimation",
  "ScrollAnimationChild",
  "ScrollAnimationParent",
  "Sentinel",
  "Slider",
  "SliderBtn",
  "SliderCount",
  "SliderDots",
  "SliderDrag",
  "SliderItem",
  "SliderProgress",
  "Sticky",
  "TableOfContent",
  "TableOfContentAnchor",
  "Tabs",
  "Transition",
]
`;
